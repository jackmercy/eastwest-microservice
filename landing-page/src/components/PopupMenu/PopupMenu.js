import React, { memo, forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './popupMenu.styles';

const PopupMenu = forwardRef(
  ({ show, className, itemClassName, list, menuPosition }, ref) => {
    const classes = useStyles();

    return show ? (
      <ul
        ref={ref}
        className={classNames(classes.menu, menuPosition, className)}
      >
        {list.map((item) => (
          <li
            data-test="popup-menu-menu-item"
            key={item.key}
            // role="presentation"
            className={classNames(classes.menuItem, itemClassName)}
            // onClick={(e) => {
            //   e.preventDefault();
            //   e.stopPropagation();
            //   console.log('????');
            // }}
            // onClick={item.onClick}
          >
            {item.label}
          </li>
        ))}
      </ul>
    ) : null;
  },
);
PopupMenu.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    }),
  ).isRequired,
  menuPosition: PropTypes.oneOf(['left', 'right']),
  show: PropTypes.bool,
  className: PropTypes.string,
  itemClassName: PropTypes.string,
};
PopupMenu.defaultProps = {
  show: false,
  menuPosition: 'left',
  className: '',
  itemClassName: '',
};

export default memo(PopupMenu);
