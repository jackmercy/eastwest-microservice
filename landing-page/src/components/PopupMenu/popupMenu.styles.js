import { createUseStyles } from 'react-jss';

import { rem } from '../../utils/jss';
import COLORS from '../../constants/COLORS';

export default createUseStyles({
  menu: {
    zIndex: 20,
    display: 'block',
    position: 'absolute',
    listStyle: 'none',
    textAlign: 'left',
    margin: 0,
    top: rem(40),
    backgroundColor: COLORS.bgWhite,
    borderRadius: rem(8),
    padding: [rem(12), 0],
    minWidth: rem(120),
    maxWidth: rem(300),
    maxHeight: rem(250),
    overflow: 'auto',
    boxShadow: '2px 4px 25px rgba(0, 0, 0, 0.25)',
    '&.left': { left: 0 },
    '&.right': { right: 0 },
  },
  menuItem: {
    textAlign: 'inherit',
    padding: [rem(8), rem(16)],
    cursor: 'pointer',
    fontWeight: 400,
    fontSize: rem(16),
    lineHeight: rem(24),
    color: COLORS.textBlack01,
    whiteSpace: 'nowrap',
    '&:hover, &.active': {
      backgroundColor: COLORS.bgGray,
      color: COLORS.primary,
    },
  },
});
