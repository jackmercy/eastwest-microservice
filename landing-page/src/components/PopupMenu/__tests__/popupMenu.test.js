import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import PopupMenu from '../PopupMenu';

describe('Testing PopupMenu component', () => {
  const props = {
    show: true,
    list: [
      { key: 'item1', label: 'Item 1' },
      { key: 'item2', label: 'Item 3' },
    ],
  };

  it('should render match snapshot', () => {
    const wrapper = shallow(<PopupMenu {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render item match with props list', () => {
    const wrapper = mount(<PopupMenu {...props} />);
    expect(wrapper.find('[data-test="popup-menu-menu-item"]')).toHaveLength(
      props.list.length,
    );
  });
});
