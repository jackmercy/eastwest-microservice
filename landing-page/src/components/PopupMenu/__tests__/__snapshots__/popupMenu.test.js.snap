// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PopupMenu component should render match snapshot 1`] = `
<ul
  className="menu-0-2-1"
>
  <li
    className="menuItem-0-2-2"
    data-test="popup-menu-menu-item"
    key="item1"
  >
    Item 1
  </li>
  <li
    className="menuItem-0-2-2"
    data-test="popup-menu-menu-item"
    key="item2"
  >
    Item 3
  </li>
</ul>
`;
